src/components/Room/Room.js

```diff
@@ -48,6 +60,317 @@ function Room() {
+      // new RoomUser's peer connection, and then setting the remote/local
+      // descriptions to complete the media acceptance agreement.
+      rssClient.awaitIncomingSDP((data) => {
+        console.log('Received incoming sdp=%o', data)
+        let peerId = data["peer_id"]
+        let pc = roomyPcs[peerId];
+        let remoteSDP = data["sdp"];
+        let desc = new RTCSessionDescription(remoteSDP);
+        let stuff = pc.setRemoteDescription(desc,
+          function() {
+            console.log('Set remote description for peerId=%o', peerId)
+            if (remoteSDP.type === "offer") {
+              console.log('Received an offer from peerId=%', peerId)
+              pc.createAnswer(
+                function(localDescription) {
+                  console.log('Answer description for peerId=%o is =%o', peerId, localDescription)
+                  pc.setLocalDescription(localDescription,
+                    function() {
+                      let sdpData = {
+                        'from_peer_id': myPeerId,
+                        'to_peer_id': peerId,
+                        'sdp': localDescription,
+                      }
+                      rssClient.relaySDP(sdpData, () => {})
+                    },
+                    function(e) {
+                      console.log('Error setting local description for peerId=%o, error=%o', peerId, e)
+                    }
+                  )
+                },
+                function(e) {
+                  console.log('error creating answer=%o', e)
+                }
+              )
+            }
+          },
+          function(e) {
+            console.log('setRemoteDescription error=%o', e)
+          }
+        )
+        console.log('description object: ', desc);
+      })
+
+      // The RFE Client must also respond to ICE or simply network connection
+      // canidate events. The ICE candidate must be added to the peer's peer
+      // connection.
+      rssClient.awaitIncomingICECandidate((data) => {
+        let peerId = data["peer_id"]
+        let pc = roomyPcs[peerId]
+        let iceCandidate = data["ice_candidate"]
+        console.log('Set ICE candidate for peerId=%o', peerId)
+        console.log(iceCandidate)
+        pc.addIceCandidate(new RTCIceCandidate(iceCandidate))
+      })
+
+      // The RFE Client must handle peers leaving the room, and delete the
+      // resources accordingly.
+      rssClient.awaitRemovePeer((data) => {
+        let peerId = data["peer_id"]
+        console.log('Removing peerId=%o from the media room', peerId)
+        if (peerId in roomyPcs) {
+          let removeStreamData = {
+            'action': 'RemoveStream',
+            'removePeerId': peerId,
+          }
+          dispatchMediaStreams(removeStreamData)
+          roomyPcs[peerId].close();
+        }
+        delete roomyPcs[peerId];
+      })
+    }) // End: joinMediaRoom.
+  }
+
+  function leaveMediaRoom() {
```

Comment:
you should add a JSDoc comment to this function too

Reply:
Also does your VS code not have JSDoc Support as described here?
https://code.visualstudio.com/docs/languages/javascript#_jsdoc-support

Reply:
thx just installed. Added